ios/Exponent/Kernel/DevSupport/EXKernelDevKeyCommands.m

```diff
@@ -68,6 +68,50 @@ - (NSString *)description
 @interface EXKernelDevKeyCommands ()
 
 @property (nonatomic, strong) NSMutableSet<EXKeyCommand *> *commands;
++ (void)handleKeyboardEvent:(UIEvent *)event;
+
+@end
+
+@interface UIEvent (UIPhysicalKeyboardEvent)
+
+@property (nonatomic) NSString *_modifiedInput;
+@property (nonatomic) NSString *_unmodifiedInput;
+@property (nonatomic) UIKeyModifierFlags _modifierFlags;
+@property (nonatomic) BOOL _isKeyDown;
+@property (nonatomic) long _keyCode;
+
+@end
+
+
+@implementation UIApplication (EXKeyCommands)
+
+- (void)EX_handleKeyUIEventSwizzle:(UIEvent *)event
+{
+  BOOL interactionEnabled = !UIApplication.sharedApplication.isIgnoringInteractionEvents;
+  BOOL hasFirstResponder = NO;
+  
+  if (interactionEnabled) {
+    UIResponder *firstResponder = nil;
+    for (UIWindow *window in [self windows]) {
+      firstResponder = [window valueForKey:@"firstResponder"];
+      if (firstResponder) {
+        hasFirstResponder = YES;
+        break;
+      }
+    }
+    
+    if (firstResponder) {
```

Comment:
it seems like possibly a safer way of handling this is to ignore the command if there is a first responder, same as in https://github.com/facebook/react-native/blob/1465c8f3874cdee8c325ab4a4916fda0b3e43bdb/React/Base/RCTKeyCommands.m#L143-L146

wdyt?

Reply:
this was my original approach but I was unable to produce a situation where RCTRootView was not the firstResponder, so it would never get called -- I suspect this is why the `r` command doesn't seem to work in RN (at least in my experience). I think the check for TextInput is a reasonable one to apply our functionality, but I'm not fully aware of all of the context / different scenarios that might cause trouble for devs.

Their implementation is similar to ours, but they always call the original method and conditionally call their extra functionality, so maybe we should do the same.

Reply:
I wonder if it would be safer to somehow put this behind a flag that could be disabled if we run into trouble with it? Is that possible?